experimental/test/test_RegularReachability.c

```diff
@@ -0,0 +1,214 @@
+        // files are enumerated from 1 to n. Thus, when loading/comparing
+        // results these values should be decremented/incremented.
+
+        // Loat graph source nodes from the sources file
+        GrB_Index s ;
+        GrB_Index S[16] ;
+        size_t ns = 0 ;
+
+        const char *name = files[k].sources ;
+        snprintf (filename, LEN, LG_DATA_DIR "%s", name) ;
+        FILE *f = fopen (filename, "r") ;
+        TEST_CHECK (f != NULL) ;
+
+        while (fscanf(f, "%ld", &s) != EOF)
+            S[ns++] = s - 1 ;
+
+        OK (fclose(f)) ;
+
+        // Load NFA starting states from the meta file
+        GrB_Index qs ;
+        GrB_Index QS[16] ;
+        size_t nqs = 0 ;
+
+        name = files[k].fa_meta ;
+        snprintf (filename, LEN, LG_DATA_DIR "%s", name) ;
+        f = fopen (filename, "r") ;
+        TEST_CHECK (f != NULL) ;
+
+        TEST_CHECK (fscanf(f, "%ld", &nqs) != EOF) ;
+
+        for (uint64_t i = 0; i < nqs; i++) {
+            TEST_CHECK (fscanf(f, "%ld", &qs) != EOF) ;
+            QS[i] = qs - 1 ;
+        }
+
+        // Load NFA final states from the same file
+        uint64_t qf ;
+        uint64_t QF[16] ;
+        size_t nqf = 0 ;
+
+        TEST_CHECK (fscanf(f, "%ld", &nqf) != EOF) ;
+
+        for (uint64_t i = 0; i < nqf; i++) {
+            TEST_CHECK (fscanf(f, "%ld", &qf) != EOF) ;
+            QF[i] = qf - 1 ;
+        }
+
+        OK (fclose(f)) ;
+
+        // Evaluate the algorithm
+        GrB_Vector r = NULL ;
+
+        OK (LAGraph_RegularReachability (&r, R, MAX_LABELS, QS, nqs,
+                                         QF, nqf, G, S, ns, msg)) ;
+
+        // Extract results from the output vector
+        GrB_Index *reachable ;
+        bool *values ;
+
+        GrB_Index nvals ;
+        GrB_Vector_nvals (&nvals, r) ;
+
+        OK (LAGraph_Malloc ((void **) &reachable, MAX_RESULTS, sizeof (GrB_Index), msg)) ;
+        OK (LAGraph_Malloc ((void **) &values, MAX_RESULTS, sizeof (GrB_Index), msg)) ;
+
+        GrB_Vector_extractTuples (reachable, values, &nvals, r) ;
+
+        // Comare the results with expected values
+        TEST_CHECK (nvals == files[k].expected_count) ;
+        for (uint64_t i = 0 ; i < nvals ; i++)
+            TEST_CHECK (reachable[i] + 1 == files[k].expected[i]) ;
+
+        // Cleanup
+        OK (LAGraph_Free ((void **) &values, NULL)) ;
+        OK (LAGraph_Free ((void **) &reachable, NULL)) ;
+
+        OK (GrB_free (&r)) ;
+
+        for (uint64_t i = 0 ; i < MAX_LABELS ; i++)
+        {
+            if (G[i] == NULL) continue ;
+            OK (LAGraph_Delete (&(G[i]), msg)) ;
+        }
+
+        for (uint64_t i = 0 ; i < MAX_LABELS ; i++ )
+        {
+            if (R[i] == NULL) continue ;
+            OK (LAGraph_Delete (&(R[i]), msg)) ;
+        }
+    }
+
+    LAGraph_Finalize (msg) ;
+}
+
+
+TEST_LIST = {
+    {"RegularReachabilityBasic", test_RegularReachabilityBasic},
```

Comment:
Naming. Look at the first comment.

Reply:
Fixed.

Reply:
Resolving, since discussed verbally Semyon Vyacheslavovich is ok.